### PR TITLE
✨ Fix links in OpenAPI description nodes

### PR DESCRIPTION
**Pull request by afranke**
Hugo already handles fixing those links for the spec website. We need to align in the OpenAPI definition generation.

Preview: https://pr3598--matrix-org-previews.netlify.app

### CODE REVIEW

**Comment by afranke** [comment]
I have copy-pasted the `baseURL` calculation from the build-spec job as I was not sure how to refactor it otherwise. Maybe I could create a job just for it and have the two build jobs that need it depend on it? I’m not sure if I can pass output around like that.

**Comment by afranke** [comment]
> I’m not sure if I can pass output around like that.

Looks like I can.

File: scripts/dump-swagger.py [hunk 1]
```diff
@@ -53,10 +53,39 @@ def resolve_references(path, schema):
...
+    elif isinstance(node, list):
+        for item in node:
+            edit_links(item, base_url)
 
 parser = argparse.ArgumentParser(
     "dump-swagger.py - assemble the Swagger specs into a single JSON file"
 )
+parser.add_argument(
+    "--base-url", "-b",
+    default="https://spec.matrix.org/latest/",
```
Comment (turt2live) [inline scripts/dump-swagger.py:85]: we should use unstable as the default as it's what we end up assuming for the rest of the build:
```suggestion
    default="https://spec.matrix.org/unstable/",
```

**Review by turt2live — COMMENTED** [review]
otherwise lgtm - thanks!

**Review by turt2live — APPROVED** [review]

**Pull request merged**